File: python chpt 3 assignments/binarytodecimal.py
def binary_to_decimal():
    binary = int(input("Enter a binary number: "))
    decimal = 0
    power = 1

    while binary != 0:
        digit = binary % 10
        decimal += digit * power
        power *= 2
        binary //= 10

    print("Decimal equivalent:", decimal)

binary_to_decimal()

File: chpt 3 assign translated to java/BinaryToDecimal.java
import java.util.Scanner;

public class BinaryToDecimal {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a binary number: ");
        int binary = scanner.nextInt();
        int decimal = 0;
        int power = 1;

        while (binary != 0) {
            int digit = binary % 10;
            decimal += digit * power;
            power *= 2;
            binary /= 10;
        }

        System.out.println("Decimal equivalent: " + decimal);
    }
}

